src/Sparrow/Allocators.Pool.cs

```diff
@@ -0,0 +1,280 @@
+﻿using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Sparrow.Binary;
+using Sparrow.Global;
+
+namespace Sparrow
+{
+    public interface IPoolAllocatorOptions : IAllocatorOptions, IComposableAllocator<Pointer>
+    {
+        int MaxBlockSize { get; }
+        int MaxPoolSizeInBytes { get; }
+    }
+
+    public static class PoolAllocator
+    {
+        public struct Default : IPoolAllocatorOptions, INativeOptions
+        {
+            public bool UseSecureMemory => false;
+            public bool ElectricFenceEnabled => false;
+            public bool Zeroed => false;
+
+            public int MaxBlockSize => 64 * Constants.Size.Megabyte;
+            public int MaxPoolSizeInBytes => 256 * Constants.Size.Megabyte;
+
+            public bool HasOwnership => true;
+
+            public IAllocatorComposer<Pointer> CreateAllocator()
+            {
+                var allocator = new Allocator<NativeAllocator<Default>>();
+                allocator.Initialize(default(Default));
+                return allocator;
+            }
+        }
+
+        public struct Static : IPoolAllocatorOptions, INativeOptions, INativeGlobalAllocatorOptions
```

Comment:
How is this different from the `Default`?

Reply:
It is expected to die when the process dies, no explicit deallocation, therefore no checking for leaks.